File: app/src/main/java/com/dennis/baseprojectandroid/application/AndroidApplication.java
package com.dennis.baseprojectandroid.application;

import android.app.Application;

import com.dennis.baseprojectandroid.core.injection.app.AppComponent;
import com.dennis.baseprojectandroid.core.injection.app.DaggerAppComponent;

/**
 * Created by BY on 2018/1/16.
 */

public class AndroidApplication extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        //Do something while initialing application.
        createAppComponent();
    }

    private void createAppComponent() {
        AppComponent appComponent = DaggerAppComponent.builder()
                .build();

        setAppComponent(appComponent);
    }

    public void setAppComponent(AppComponent appComponent) {
        this.appComponent = appComponent;
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/kotlin/com/dennis/baseprojectandroid/application/AndroidApplication.kt
package com.dennis.baseprojectandroid.application

import android.app.Activity
import android.app.Application
import com.dennis.baseprojectandroid.core.injection.app.AppInjector
import com.dennis.baseprojectandroid.core.plugins.setupFabric
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import javax.inject.Inject

/**
 * Created by BY on 2018/1/16.
 */

class AndroidApplication : Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    override fun onCreate() {
        super.onCreate()

        //Do something while initialing application.
        AppInjector.init(this)

        setupFabric(this)
    }

    override fun activityInjector(): AndroidInjector<Activity> = dispatchingAndroidInjector
}
